### PR TITLE
Warning rework

### PR DESCRIPTION
**Pull request by PaulJonasJost**
Fixing various warnings.

Currently reworked:
* Problem reading -> as it always creates warnings
* Matplotlib error with set_yticklabels()
* should remove `distplot` warning from seaborn
* added comment where I think an error of the `axis` argument error in seaborn is.
* 348 of the 440 warnings I saw were from the theano/tensor/basic.py`as it was still using np.complex... nothing we can do there
* WARNING - Parameters obtained from history and optimizer do not match: [0.99677274 0.99999951], [0.55488232 0.31315563] will be handled by other PR
* removed `DeprecationWarning: Flags not at the start of the expression '^(?i)(ls_)'`

Summary:
* base: 440 -> 358 of which 348 are from theano
* petab: should remove the only warning when amici merges develop into master
* optimize: 18-> 17 + needs discussion on 'ftol'

### CODE REVIEW

File: pypesto/visualize/sampling.py [hunk 1]
```diff
@@ -977,6 +977,7 @@ def sampling_scatter(
 
     sns.set(style="ticks")
 
+    # TODO: Think this throws the axis errors in seaborn.
```
Comment (yannikschaelte) [inline pypesto/visualize/sampling.py:980]: TODO Now or TODO Later? If later, maybe "TODO PJ" so we know whom to ask ^^

File: pypesto/visualize/sampling.py [hunk 2]
```diff
@@ -1045,12 +1046,28 @@ def sampling_1d_marginals(
     # fig, ax = plt.subplots(nr_params, figsize=size)[1]
     for idx, par_id in enumerate(param_names):
         if plot_type == 'kde':
-            sns.kdeplot(params_fval[par_id], bw=bw, ax=par_ax[par_id])
+            # bw_method argument is deprecated... FutureWarning:
```
Comment (yannikschaelte) [inline pypesto/visualize/sampling.py:1049]: bw_method is deprecated in favor of bw_method?

File: test/optimize/test_optimize.py [hunk 3]
```diff
@@ -259,8 +260,9 @@ def test_mpipoolengine():
              f'{path}/../../doc/example/example_MPIPool.py'])
 
         # read results
-        opt_result_reader = OptimizationResultHDF5Reader('temp_result.h5')
-        result1 = opt_result_reader.read()
+        result1 = read_result('temp_result.h5',
+                              problem=True,
+                              optimize=True)
```
Comment (yannikschaelte) [inline test/optimize/test_optimize.py:265]: totally random remark: this is the style preferred in the python community. obviously no problem though

```suggestion
        result1 = read_result(
            'temp_result.h5',
            problem=True,
            optimize=True,
        )
```

**Review by yannikschaelte — APPROVED** [review]
looks good to me, great!

File: test/optimize/test_optimize.py (hunk 3 above)
Comment (PaulJonasJost) [reply test/optimize/test_optimize.py:265]: no that is good to hear as it is the style I like more. I actually thought this is the one the python community does not like as it creates more lines. thanks 🙏

**Comment by yannikschaelte** [comment]
@PaulJonasJost what is the status on this?

**Comment by PaulJonasJost** [comment]
> @PaulJonasJost what is the status on this?

@yannikschaelte there is still one thing missing which perhaps requires a discussion. But I think I will do a separate PR for this.

**Pull request merged**